src/RoadmapTopic/RoadmapTopic.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { Lesson } from "models";
+import { RoadmapTopicContainer, RoadmapTopicTitle } from "./RoadmapTopic.styled";
+import { AppColors, Colors } from "styles/types";
+import { capitalizeFirstLetter } from "utils";
+import Popover from "molecules/Popover/Popover";
+import RoadmapLesson from "RoadmapLesson/RoadmapLesson";
+import Button from "atoms/Button/Button";
+
+interface RoadmapTopicProps {
+  name: string;
+  trackName?: string;
+  lessons: Lesson[];
+}
+const RoadmapTopic: React.FC<RoadmapTopicProps> = ({ name, lessons, trackName }) => {
+  return (
+    <RoadmapTopicContainer>
+      <RoadmapTopicTitle>{capitalizeFirstLetter(name)}</RoadmapTopicTitle>
+      {lessons.map((lesson, i) =>
+        lesson.wip ? (
```

Comment:
a lot ternaries 😅 WDYT about moving the "return" components above the "parent" one or to some outside file? For sure it'll improve readability 🙂

Reply:
There is one 😅 and it's not loaded at all (the "return" JSX elements are already components defined elsewhere). For a 47 lines component, readability should not be a problem ;)